lib/Dialect/TritonGPU/Transforms/OptimizeThreadLocality.cpp

```diff
@@ -122,7 +122,7 @@ class TritonGPUOptimizeThreadLocalityPass
         return;
       for (auto operand : reduce->getOperands()) {
         auto def = operand.getDefiningOp();
-        if (!isa<triton::LoadOp>(def))
+        if (def == nullptr || !isa<triton::LoadOp>(def))
```

Comment:
nit: you can do directly: `operand.getDefintion<triton::LoadOp>`
if not you can also do isa_and_notnull